### PR TITLE
Migration: GitHub Actions

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
- Add GitHub Actions for CI and CD (based off the functionality
  from the Engine)
- Add Issue Templates for bugs and features
- Add links to the communities
- Add a pull request template

### CODE REVIEW

**Comment by BenjamenMeyer** [comment]
This covers master after we've gotten this done then we can port to satisfy #43.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Thank you for submitting a pull request and becoming a contributor to the Vega Strike: Upon the Coldest Sea.
```
Comment (stephengtuggy) [inline .github/pull_request_template.md:1]: Omit `the` before `Vega Strike`

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
Looks good overall; just a few things we might want to change

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+contact_links:
+  - name: Ask the Community on Gitter
+    url: https://gitter.im/vegastrike/community
+    about: Join the Vega Strike Gitter Discussion
+  - name: Vega Strike Forums
+    url: https://forums.vega-strike.org/
+    about: Join the Vega Strike Forums
+  - name: Vega Strike Official Wiki
+    url: https://wiki.vega-strike.org/
+    about: Read up at the Vega Strike Wiki
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/ISSUE_TEMPLATE/config.yml:11]: yeah...not sure how to include mailing lists in there yet so I'm going to wait on those. Probably just use the `mailto://` URL format. I'm going to leave that to a follow-on PR.

File: .github/workflows/gh-actions-pr.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          #- FROM:     'ubuntu:bionic'
+          #  COMPILER: 'gcc'
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          #- FROM:     'ubuntu:bionic'
+          #  COMPILER: 'clang'
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          #- FROM:     'ubuntu:xenial'
+          #  COMPILER: 'gcc'
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-pr.yml:29]: I was matching the old Travis-CI config which only did one. It's not as big a deal on the Assets side as that's mostly Python. I'm game either way though.

File: .github/workflows/gh-actions-release.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          #- FROM:     'ubuntu:bionic'
+          #  COMPILER: 'clang'
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          #- FROM:     'ubuntu:xenial'
+          #  COMPILER: 'gcc'
+          #  FLAGS:    '-DUSE_PYTHON_3=ON'
+          - FROM:     'debian:buster'
+            COMPILER: 'clang'
+            FLAGS:    '-DUSE_PYTHON_3=ON'
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-release.yml:30]: prefer Buster (Debian 10) over Stretch (Debian 9) (https://wiki.debian.org/DebianReleases)

File: .github/workflows/gh-actions-pr.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (stephengtuggy) [reply .github/workflows/gh-actions-pr.yml:29]: We can leave it as is for now.

File: .github/workflows/gh-actions-release.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (stephengtuggy) [reply .github/workflows/gh-actions-release.yml:30]: > 
> 
> prefer Buster (Debian 10) over Stretch (Debian 9) (https://wiki.debian.org/DebianReleases)

Understood. In general, yes, that is true: Debian Buster is preferred over Stretch. However, when it comes to building .debs, I think it's best to build on the oldest .deb-based platform we support. That way, we don't have to worry about backward compatibility of the .deb format, or any system dependencies, etc.

[the remark this answers is not in the document]
Comment (stephengtuggy) [reply .github/workflows/gh-actions-release.yml:30]: So do we still support Debian stretch at this point?

[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-release.yml:30]: I'll have to double-check but iirc we do

[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-release.yml:30]: Okay checked (and updated) https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Platform-Support and we do still support Stretch for the time being as its the Debian LTS release. Buster is Debian Current. I think we have about 9 months left before LTS/Current change up based on https://wiki.debian.org/DebianReleases.

[the remark this answers is not in the document]
Comment (stephengtuggy) [reply .github/workflows/gh-actions-release.yml]: OK

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/ISSUE_TEMPLATE/config.yml:11]: @stephengtuggy added something using https://mailtolinkgenerator.com/ to generate the full link.

File: .github/workflows/gh-actions-release.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-release.yml]: Changed from Buster to Stretch; we'll probably update it to Buster in about 6-9 months

**Review by stephengtuggy — APPROVED** [review]
LGTM!

**Pull request merged**